### PR TITLE
Feat/idp implement

### PR DESCRIPTION
**Pull request by GiaBaorr**
## Release Notes

- **New Features**
	- Introduced user signup and login functionality with dedicated pages.
	- Added internationalization support with English and Vietnamese translations.
	- Implemented enhanced error handling with a centralized exception handler.
	- Integrated Tailwind CSS for improved styling across the application.
	- Added support for user role management with defined roles.
	- Introduced a new configuration for managing environment variables.

- **Bug Fixes**
	- Updated email validation regex for improved accuracy.
	- Enhanced password security by switching to BCryptPasswordEncoder.

- **Documentation**
	- Added detailed messages for user authentication and registration processes.

- **Chores**
	- Updated `.gitignore` to exclude unnecessary files.
	- Added a new `docker-compose.yml` configuration for service renaming.

### CODE REVIEW

File: sep490-idp/src/main/java/sep490/idp/configs/I18nConfig.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        localeResolver.setDefaultLocale(Locale.ENGLISH);
+        return localeResolver;
+    }
+
+    @Bean
+    public LocaleChangeInterceptor localeChangeInterceptor() {
+        LocaleChangeInterceptor interceptor = new LocaleChangeInterceptor();
+        interceptor.setParamName("lang");
+        return interceptor;
+    }
```
[the remark this answers is not in the document]
Comment (GiaBaorr) [reply sep490-idp/src/main/java/sep490/idp/configs/I18nConfig.java:29]: useless props

File: sep490-common/src/main/java/sep490/common/api/utils/EnumUtil.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    private EnumUtil() {
+        // Utility class, no instances allowed
+    }
+
+    public static <T extends BaseEnum> T getCodeFromString(Class<T> clazz, String code) {
+        return Arrays.stream(clazz.getEnumConstants())
+            .filter(e -> StringUtils.equals(e.getCode(), code))
+            .findFirst()
+            .orElse(null);
+    }
```
[the remark this answers is not in the document]
Comment (GiaBaorr) [reply sep490-common/src/main/java/sep490/common/api/utils/EnumUtil.java:19]: Return null is better, throwing runtimeexception is more dangerous

File: .run/IdentityProvider.run.xml [hunk 3]
```diff
@@ -2,9 +2,9 @@
   <configuration default="false" name="IdentityProvider" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot" nameIsGenerated="true">
     <envs>
       <env name="DB_HOST" value="localhost" />
-      <env name="DB_NAME" value="sep490" />
+      <env name="DB_NAME" value="sep490_idp" />
       <env name="DB_PORT" value="5432" />
-      <env name="DB_SCHEMA" value="idp" />
+      <env name="DB_SCHEMA" value="public" />
```
[the remark this answers is not in the document]
Comment (GiaBaorr) [reply .run/IdentityProvider.run.xml:7]: default is public, no need to update

File: sep490-idp/src/main/resources/templates/signup.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            <div class="text-center mt-4">
+                <p class="text-gray-700">
+                    <span th:text="#{signup.label.existingAccount}"></span>
+                    <a th:href="@{/login}" class="text-green-500 hover:underline" th:text="#{signup.btn.login}"></a>
+                </p>
+            </div>
+            <div class="text-center mt-3">
+                <a th:href="@{''(lang=en)}" class="text-green-500 hover:underline" th:text="#{language.english}">English</a> |
+                <a th:href="@{''(lang=vi)}" class="text-green-500 hover:underline" th:text="#{language.vietnamese}">Vietnamese</a>
+            </div>
```
[the remark this answers is not in the document]
Comment (GiaBaorr) [reply sep490-idp/src/main/resources/templates/signup.html:81]: @thongdanghoang @nganntqe170236 @huynhlephcvinh   should this suggestion applied?
Store form data (except password) to improve UX?

**Review by thongdanghoang — APPROVED** [review]

**Pull request merged**